air-quality-backend/src/air_quality/etl/forecast/forecast_adapter.py

```diff
@@ -51,44 +53,54 @@ def _get_overall_aqi_value_for_slice(
     )
 
 
-def transform(forecast_data: ForecastData, location: AirQualityLocation) -> list:
+def transform(forecast_data: ForecastData, locations: list[AirQualityLocation]) -> list:
+    pollutant_data_with_location = forecast_data.get_pollutant_data_for_locations(
+        locations, list(PollutantType)
+    )
+    logging.debug(f"Pollutant data with location: {len(pollutant_data_with_location)}")
+
     valid_time_values = forecast_data.get_valid_time_values()
     step_values = forecast_data.get_step_values()
     model_base_time = datetime.utcfromtimestamp(forecast_data.get_time_value())
     pollutant_forecast_for_location = []
-    location_name = location["name"]
-    location_type = location["type"]
-    logging.debug(f"Processing location: {location_name} ({location['type']})")
-    forecast_data_by_type = _get_pollutant_values_by_type(forecast_data, location)
-
-    for i in range(0, valid_time_values.size):
-        measurement_timestamp = valid_time_values[i]
-        measurement_date = datetime.utcfromtimestamp(measurement_timestamp)
-        overall_aqi_value = _get_overall_aqi_value_for_slice(forecast_data_by_type, i)
 
-        pollutant_data = {}
-        for pollutant_type in PollutantType:
-            forecast_data = forecast_data_by_type[pollutant_type]
-            pollutant_data[pollutant_type.value] = {
-                "aqi_level": forecast_data["aqi_values"][i],
-                "value": forecast_data["values_ug_m3"][i],
-            }
-
-        document = {
-            "name": location_name,
-            "location_type": location_type,
-            "location": {
-                "type": "Point",
-                "coordinates": [location["longitude"], location["latitude"]],
-            },
-            "forecast_base_time": model_base_time,
-            "forecast_valid_time": measurement_date,
-            "forecast_range": int(step_values[i]),
-            "overall_aqi_level": overall_aqi_value,
-            "source": "cams-production",
-            **pollutant_data,
+    for location, data_by_pollutant in pollutant_data_with_location:
+        location_name = location["name"]
+        location_type = location["type"]
+        logging.debug(f"Processing location: {location_name} ({location['type']})")
+        forecast_data_by_type = {
+            pollutant_type: _convert_pollutant_values(pollutant_type, data)
+            for pollutant_type, data in data_by_pollutant.items()
         }
 
-        pollutant_forecast_for_location.append(document)
+        for i in range(0, step_values.size):
+            measurement_timestamp = valid_time_values[i]
+            measurement_date = datetime.utcfromtimestamp(measurement_timestamp)
```

Comment:
vscode suggests this way of creating a utc datetime is deprecated, something to do with naive datetimes. More details [here ](https://docs.python.org/3/library/datetime.html#datetime.datetime). Suggested way is as below
```suggestion
            measurement_date = datetime.fromtimestamp(measurement_timestamp, timezone.utc)
```

Reply:
Lovely - I've changed everywhere